src/toastfile.rs

```diff
@@ -12,6 +13,67 @@ pub const DEFAULT_LOCATION: &str = "/scratch";
 // The default user for commands and files copied into the container
 pub const DEFAULT_USER: &str = "root";
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct MappingPath {
+    pub host_path: PathBuf,
+    pub container_path: PathBuf,
+}
+
+impl Serialize for MappingPath {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(
+            format!(
+                "{}:{}",
+                self.host_path.to_str().unwrap(),
+                self.container_path.to_str().unwrap(),
+            )
+            .as_str(),
+        )
+    }
+}
+
+struct MappingPathVisitor;
+
+impl<'de> serde::de::Visitor<'de> for MappingPathVisitor {
+    type Value = MappingPath;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        formatter.write_str("cannot parse mapping path")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        if let Some((host_path, container_path)) = v.split_once(":") {
+            Ok(MappingPath {
+                host_path: host_path.to_owned().parse().expect("illegal host path"),
```

Comment:
Does this panic if the path cannot be parsed? Can we return an `Err` instead?

(Same concern about the other `expect` calls below.)